### PR TITLE
Add new endpoints for upgrade RabbitMQ and Erlang

### PR DESCRIPTION
**Pull request by tbroden84**
Two new endpoints to 

- Retrieve latest possible upgradable versions for RabbitMQ and Erlang
- Invoke upgrade of RabbitMQ and Erlang versions

Co-authored-by: Johan Nordlund <johan.n@84codes.com>

### Friendly reminders
- [ ] Describe the problem / feature (ideally with [meaningful commit messages](https://tekin.co.uk/2019/02/a-talk-about-revision-histories))
- [ ] Lint rules pass
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] The environment (`heroku config`) has been updated if needed (new `ENV` variables)

### Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: api/upgrade_rabbitmq.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	if response.StatusCode == 200 {
+		return "Already at highest possible version", nil
+	} else if response.StatusCode != 202 {
+		return "", fmt.Errorf("Upgrade RabbitMQ failed, status: %v, message: %s", response.StatusCode, failed)
+	}
+
+	return api.waitUntilUpgrade(instanceID)
+}
+
+func (api *API) waitUntilUpgrade(instanceID int) (string, error) {
```
Comment (dentarg) [inline api/upgrade_rabbitmq.go:44]:
```suggestion
func (api *API) waitUntilUpgraded(instanceID int) (string, error) {
```

Comment (dentarg) [reply api/upgrade_rabbitmq.go:44]: or `waitUntilUpgradeComplete` or `waitUntilUpgradeDone` or `waitUntilUpgradeFinished` :)

File: api/upgrade_rabbitmq.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if err != nil {
+		return "", err
+	}
+	if response.StatusCode == 200 {
+		return "Already at highest possible version", nil
+	} else if response.StatusCode != 202 {
+		return "", fmt.Errorf("Upgrade RabbitMQ failed, status: %v, message: %s", response.StatusCode, failed)
+	}
+
+	return api.waitUntilUpgrade(instanceID)
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:41]:
```suggestion
	return api.waitUntilUpgraded(instanceID)
```

File: api/upgrade_rabbitmq.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+func (api *API) waitUntilUpgrade(instanceID int) (string, error) {
+	var data []map[string]interface{}
+	failed := make(map[string]interface{})
+
+	for {
+		time.Sleep(30 * time.Second)
+		path := fmt.Sprintf("api/instances/%v/nodes", instanceID)
+		_, err := api.sling.New().Path(path).Receive(&data, &failed)
+		if err != nil {
+			log.Printf("[ERROR] go-api::upgrade_rabbitmq::waitUntilUpgrade error: %v", err)
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:53]:
```suggestion
			log.Printf("[ERROR] go-api::upgrade_rabbitmq::waitUntilUpgraded error: %v", err)
```

File: api/upgrade_rabbitmq.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+			return "", err
+		}
+		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgrade numberOfNodes: %v", len(data))
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:56]:
```suggestion
		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgraded numberOfNodes: %v", len(data))
```

File: api/upgrade_rabbitmq.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgrade data: %v", data)
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:57]:
```suggestion
		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgraded data: %v", data)
```

File: api/upgrade_rabbitmq.go [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+		ready := true
+		for _, node := range data {
+			log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgrade ready: %v, configured: %v",
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:60]:
```suggestion
			log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgraded ready: %v, configured: %v",
```

File: api/upgrade_rabbitmq.go [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+				ready, node["configured"])
+			ready = ready && node["configured"].(bool)
+		}
+		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgrade ready: %v", ready)
```
Comment (tbroden84) [inline api/upgrade_rabbitmq.go:64]:
```suggestion
		log.Printf("[DEBUG] go-api::upgrade_rabbitmq::waitUntilUpgraded ready: %v", ready)
```

**Pull request merged**